src/DynamoCore/Scheduler/UpdateGraphAsyncTask.cs

```diff
@@ -77,7 +77,12 @@ internal bool Initialize(EngineController controller, WorkspaceModel workspace)
                 {
                     var node = workspace.Nodes.FirstOrDefault(n => n.GUID.Equals(nodeGuid));
                     if (node != null)
+                    {
                         node.ClearDirtyFlag();
+
+                        // Set IsEvaluating to true, so that node's preview bubble won't crash.
+                        node.IsEvaluating = true;
```

Comment:
Hmmm... I  just read comments near `CachedValue` and they scared me :)

From these summaries I understood, that it's dangerous to set MirrorData not trough scheduler and engine...

![image](https://cloud.githubusercontent.com/assets/8158404/13492330/12f09ab8-e140-11e5-9ad0-863a4093b333.png)
![image](https://cloud.githubusercontent.com/assets/8158404/13492364/406dcf4c-e140-11e5-9757-bf3211b7988f.png)
![image](https://cloud.githubusercontent.com/assets/8158404/13492381/5de92d78-e140-11e5-8013-1d4331d083e6.png)